app/src/main/java/calculator/controller/MainController.java

```diff
@@ -0,0 +1,46 @@
+package calculator.controller;
+
+import static calculator.entity.Menu.getSelectedMenu;
+import static calculator.view.InputView.inputMenuNumber;
+import static calculator.view.OutputView.showMenu;
+
+import calculator.entity.Menu;
+import calculator.service.Calculator;
+import calculator.service.HistoryReader;
+
+public class MainController {
+
+    private final HistoryReader historyReader;
+    private final Calculator calculator;
+
+    public MainController(HistoryReader historyReader, Calculator calculator) {
+        this.historyReader = historyReader;
+        this.calculator = calculator;
+    }
+
+    public void run() {
+        while (true) {
+            showMenu();
+
+            int menuNumber = inputMenuNumber();
+            Menu selectedMenu = getSelectedMenu(menuNumber);
+
+            if (selectedMenu.isQuit()) {
+                break;
+            }
+
+            execute(selectedMenu);
+        }
```

Comment:
최대한 영주님의 코드 스타일대로 유지한다면 이러한 코드 제안은 어떨까요?
영주님이 고민하시는 while 문을 활용하는것에 초점을 맞추어보았어요.
execute() 는 내부적으로 메뉴에 의해 적절한 수행을 하기에 문제가 되지 않을것 같아요.

```suggestion
    public void run() {
    	boolean isRunning = true;
    	
        while (isRunning) {
            showMenu();

            int menuNumber = inputMenuNumber();
            Menu selectedMenu = getSelectedMenu(menuNumber);
            
            isRunning = !selectedMenu.isQuit();

            execute(selectedMenu);
        }
```

Reply:
와...저는 왜 이런 생각을 못했을까요! 정말 감사합니다 ㅎㅎㅎ